File: src/main/java/com/zp4rker/core/discord/command/Command.java
package com.zp4rker.core.discord.command;

import net.dv8tion.jda.core.Permission;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * The command @interface.
 *
 * @author zpdev
 */
@Retention(RetentionPolicy.RUNTIME)
public @interface Command {

    String[] aliases();
    //String description() default "";
    String usage() default "";

    Permission permission() default Permission.MESSAGE_READ;
    long role() default 0;

    int args() default 0;
    int minArgs() default 0;
    int mentionedMembers() default 0;
    int mentionedChannels() default 0;
    int mentionedRoles() default 0;

    boolean autodelete() default false;

    //boolean hidden() default false;

}

File: src/main/kotlin/com/zp4rker/core/discord/command/Command.kt
package com.zp4rker.core.discord.command

import net.dv8tion.jda.api.Permission

abstract class Command(
        val aliases: Array<String>,
        val description: String = "",
        val usage: String = aliases[0],

        val permission: Permission = Permission.MESSAGE_READ,
        val role: Long = 0,

        val args: Int = 0,
        val minArgs: Int = 0,
        val mentionedMembers: Int = 0,
        val mentionedRoles: Int = 0,
        val mentionedChannels: Int = 0,

        val autoDelete: Boolean = false
)